README.md

```diff
@@ -3,61 +3,149 @@
+PDE simulations. The intent is to abstract away from details such as
+files and formats and describe the simulation as one or more
+discretizations of a manifold, complete with tangent space,
+fiber-bundles, and sub-manifolds. The project is currently in a
+working beta.
+
+More information can be found in
+[this presentation](https://github.com/Yurlungur/simulationio-and-yt)
+on SimulationIO and related tools.
+
+## Components
+
+SimulationIO has several components:
+- A collection of object files that can be linked against. For now,
+  only static linking is supported.
+- Several utility programs that can, for example, convert HDF5 output
+  from the [Einstein Toolkit](http://einsteintoolkit.org/) into the
+  SimulationIO format.
+- A python library, pysimulationio that wraps the object files into a
+  single shared python API.
+
+## Using
+
+- To convert output from the Einstein Toolkit:
+
+```bash
+sio-convert-carpet-output output_file.s5 /path/to/input_files/*.h5
+```
+
+- To look at the innards of a file:
+
+```bash
+sio-list filename.s5
+```
+
+- To generate an example file:
+
+```bash
+sio-example
+```
+
+- The repository also contains several python files which show how to
+  use the python API.
+
+## Requirements
+
+SimulationIO relies on a modern version of HDF5 and a modern version
+of MPI.
+
+## Installing
+
+The python library depends on but is nequired for the rest of the
+library. Therefore, it is installed separately.
+
+### Installng the core library
+
+To install the utilities and object files, clone the repository. Then
+set your environment variables so that the make system can find your
+HDF5 and MPI libraries. There are defaults set, but they may not work
+for you. The following environment variables are supported (and in
+this case set to the default):
+
+```bash
+CXX=g++
+```
+
+You may use CXX to set a compiler other than g++, for example the
+intel compilers.
+
+```bash
+MPI_NAME=mpich
+MPI_DIR=/usr
+MPI_LIBS="-lmpichcxx -lmpich"
+```
+
+This assumes that the mpi include directory is `${MPI_DIR}/include`
+and lib directory is `${MPI_DIR}/lib/x86_64-gnu`. You may also
+explicitly set these values:
+
+```bash
+MPI_INCLUDE=${MPI_DIR}/include/${MPI_NAME}
+MPI_CPPFLAGS=-I${MPI_INCLUDE}
+MPI_CXXFLAGS=
+MPI_LINK=${MPI_DIR}/lib/x86_64-gnu
+MPI_LDFLAGS="-L${MPI_LINK} -Wl,-rpath,${MPI_LINK}"
+```
+
+Similarly, for HDF5:
+
+```bash
+HDF5_DIR=/usr/local/hdf5
+HDF5_INCLUDE=${HDF5_DIR}/include
+HDF5_CPPFLAGS=-I{HDF5_INCLUDE}
+HDF5_CXXFLAGS=
+HDF5_LINK=${HDF5_DIR}/lib
+HDF5_LDFLAGS="-L${HDF5_LINK} -Wl,-rpath,${HDF5_LINK}"
+HDF5_LIBS="-lhdf5_cpp -lhdf5"
+```
+
+Once you've set your environment variables, run
+
+```bash
+make -j N
+```
+
+where `N` is the number of processors you'd like to use. This
+generates the object files and utilities. If you'd like to install the
+utilities in a global location you can set that location with the
+shell variables
+
+```bash
+PREFIX=/usr/local
+BIN_DIR=bin
```

Comment:
This wouldn't usually be called `BIN_DIR` since it is appended to another directory. Maybe `BIN_SUBDIR`? But it's basically always `bin`, so you could also leave it out.